src/ember-app.js

```diff
@@ -205,10 +208,14 @@ class EmberApp {
    *                                       only
    */
   async getNewApplicationInstance() {
+    const appInstanceRetrievalStartTime = Date.now();
     const queueObject = this._sandboxApplicationInstanceQueue.dequeue();
     const app = await queueObject.item;
 
-    return this.getAppInstanceInfo(app, queueObject.isItemPreBuilt);
+    return this.getAppInstanceInfo(app, {
+      isAppInstancePreBuilt: queueObject.isItemPreBuilt,
+      appInstanceRetrievalTime: Date.now() - appInstanceRetrievalStartTime,
```

Comment:
I'd like to store the start and stops too. Also can we rename this to `buildAppInstanceTiming`?

Something like:
```js
const start = performance.now();
const queueObject = this._sandboxApplicationInstanceQueue.dequeue();
const app = await queueObject.item;
const stop = performance.now();

return this.getAppInstanceInfo(app, {
  isAppInstancePreBuilt: queueObject.isItemPreBuilt,
  buildAppInstanceTiming: {
    start,
    stop,
    duration: stop - start,
  }
});
```

Reply:
Rob,

Do we need all 3 of 'start', 'stop', and 'duration'? 

I'd prefer either (1) 'start' and 'stop' or (2) 'start' and 'duration'. I prefer the latter option more. The reason I like it is because: (1) it's smaller (which isn't a huge deal) and (2) whenever I see all 3 I do the subtraction to make sure 'duration' = 'stop' - 'start' because I always ask myself, why would they provide all 3?

That being said, this is not such a strong opinion and if you are strongly opinionated, I'll defer.

Thanks,
Mark

Reply:
My reasoning here is _basically_ that "clock time duration" isn't enough information in all cases, but in many cases it is. For example, if we just provided duration it is actually quite difficult to map the results to see how many were interleaved/overlapping/whateverTheWordIs. In general, I feel like this is in a "why not" kinda scenario (where it costs us roughly nothing to do all three, and in fact we _basically_ have to know all three anyways in order to provide a duration). That being said, I definitely do **not** feel strongly about providing all 3. I think `start` and `duration` is probably enough info (and addresses my primary issue).

Let's call it "dealers choice", to the PR author go the spoils 😝

Reply:
lol okk.... i will unveil the suspense in my next commit  ;)

Reply:
FWIW [`PerformanceEntry`](https://nodejs.org/api/perf_hooks.html#perf_hooks_class_performanceentry) defines `startTime` and `duration`